NitroxPatcher/Patches/Dynamic/ItemsContainer_NotifyAddItem_Patch.cs

```diff
@@ -10,12 +10,18 @@ public class ItemsContainer_NotifyAddItem_Patch : NitroxPatch, IDynamicPatch
     {
         public static readonly Type TARGET_CLASS = typeof(ItemsContainer);
         public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("NotifyAddItem", BindingFlags.NonPublic | BindingFlags.Instance);
+        private static ItemContainers itemContainers;
 
         public static void Postfix(ItemsContainer __instance, InventoryItem item)
         {
+            if (itemContainers == null)
+            {
+                itemContainers = NitroxServiceLocator.LocateService<ItemContainers>();
+            }
```

Comment:
Don't these already cache internally?  Also, will this hang on to a dead reference if we refresh the service locator (such as when reconnecting)?

Reply:
I don't think they do because had this once inside the protobuf patches and it slowed the game down to a crawl. I'll check the decompiled code now to make sure.

Reply:
Nvm, it does cache internally but it does some casts and checks. I'll revert the changes in methods that aren't called so often.